### DISCUSSION TITLE
Superfluous :GRAY assignment in depth first visit?

### DISCUSSION THREAD

**Discussion by delphaber** [monora/rgl#65] [Q&A] [answered]
Hello!

Isn't line 200 superfluous? We are already marking the vertex as :GRAY in line 192 when we go deeper in the recursion level at line 201.

Just wondering if I'm missing something :)

https://github.com/monora/rgl/blob/3cdf5c24b3d3d0368907b4ba9e6af3072e4b5e62/lib/rgl/traversal.rb#L191-L201

**Comment by monora** [comment]
On line 192 vertex `u` is marked :GRAY, whereas on line 200 vertex `v`, which is adjacent to `u` is also marked :GRAY, because the edge `(u,v)` shall be followed (`vis.follow_edge(u, v)` is true).

**Reply by delphaber** [reply]
Thanks for the answer but I still don't get it, sorry 😅 
Vertex `v` passed to the block in line 195 will become vertex `u` when calling depth_first_visit recursively in line 201, and it will be coloured (again) to :GRAY in line 192.

**Reply by monora** [reply]
You are right! Thank you for this thorough code review! Deleting this 10 year old line does not break the tests! Issue #66 will fix it.

**Reply by delphaber** [reply, marked as answer]
Good! My graph teory study is rusty so I was not sure if I was missing something! Thanks for this gem!
